payments-model/src/main/java/com/stripe/android/model/ConsumerPaymentDetails.kt

```diff
@@ -26,22 +26,34 @@ data class ConsumerPaymentDetails(
     data class Card(
         override val id: String,
         override val last4: String,
-    ) : PaymentDetails(id, type = "card")
+    ) : PaymentDetails(id, type = TYPE) {
+        companion object {
+            const val TYPE = "card"
```

Comment:
why is this necessary? I would expect that you could just access type directly without needing this new val

Reply:
I need a set of the paymentTypes [here](https://github.com/stripe/stripe-android/blob/8808cbb9ff0bbe64f8371cefbc8512e5d137666d/link/src/main/java/com/stripe/android/link/model/SupportedPaymentMethodTypes.kt#L24). I would need to create an instance of each payment detail to obtain the type with the current implementation. So I moved it to a companion object, similar to [how it was](https://github.com/stripe/stripe-android/blob/3d373fc3e46053b9efde08f29b50d0720ef10e71/payments-core/src/main/java/com/stripe/android/model/ConsumerPaymentDetails.kt#L30) before link was removed.